source/_cookbook/SSL guide.md

```diff
@@ -0,0 +1,570 @@
+</p>
+
+now SSH in to the device your HA is running on.
+
+
+<p class='note'>
+If you're running the 'standard' setup on a raspberry pi the chances are you just logged in as the 'pi' user.  If not, you may have logged in as the HA user.  There are commands below that require the HA user to be on the sudoers list.  If you are not using the 'standard' pi setup it is presumed you will know how to get your HA user on the sudoers list before continuing.  If you are running the 'standard' pi setup, from your 'pi' user issue the following command (where <hass> is the HA user):
+
+```bash
+$ sudo adduser <hass> sudo
+```
+
+</p>
+
+If you did not already log in as the user that currently runs HA, change to that user (usually hass or homeassistant - you may have used a command similar to this in the past):
+
+```bash
+$ su -s /bin/bash hass 
+```
+
+Make sure you are in the home directory for the HA user:
+
+```bash
+$ cd
+```
+
+We will now make a directory for the certbot software, download it and give it the correct permissions:
+
+```text
+$ mkdir certbot
+$ cd certbot/
+$ wget https://dl.eff.org/certbot-auto
+$ chmod a+x certbot-auto
+```
+
+Now we will run the certbot program to get our ssl certificate.  You will need to include your email address and your duckDNS url in the appropriate places:
+
+```text
+$ ./certbot-auto certonly --standalone --preferred-challenges http-01 --email your@email.address -d examplehome.duckdns.org
+```
+
+Once the program has run it will generate a certificate and other files and place them in a folder `/etc/letsencrypt/` .
+
+Confirm this file has been populated:
+
+```bash
+$ ls /etc/letsencrypt/live/
+```
+
+This should show a folder named exactly after your DuckDNS url.
+
+Our HA user needs access to files within the letsencrypt folder, so issue the following commands to change the permissions.
+
+```bash
+$ sudo chmod 755 /etc/letsencrypt/live/
+$ sudo chmod 755 /etc/letsencrypt/archive/
+```
+
+Did all of that go without a hitch?  Wahoo! Your LetsEncrypt certificate is now ready to be used with home assistant.  Move to step 5 to put it all together...
+
+#### {% linkable_title 5 - check the incoming conection %}
+
+<p class='note'>
+Following on from Step 4 your SSH will still be in the certbot folder.  If you edit your configuration files over SSH you will need to change to your homeassistant folder:
+
+```bash
+$ cd ~/.homeassistant
+```
+
+If you use samba shares to edit your files you can exit your SSH now.
+</p>
+
+Go to your router's configuration pages and set up a new port forwarding rule, thus:
+
+```text
+Service name - ha_ssl
+Port Range - 443
+Local IP - YOUR-HA-IP
+Local Port - 8123
+Protocol - Both
+```
+
+If during step 4 you had to use port 443 instead of port 80 to generate your certificate, you should delete that rule now.

```

Comment:
Maybe rewrite or reorder to line 301 to make clear which rule you mean with "delete that rule"